### PR TITLE
Conver init args to functional options

### PR DESCRIPTION
**Pull request by twoism**
This change fixes the breaking change introduced with AllowDotFiles while preserving the API.

### CODE REVIEW

File: loader/loader.go [hunk 1]
```diff
@@ -157,12 +157,12 @@ func getFileSystemOp(ev fsnotify.Event) FileSystemOp {
 	return -1
 }
 
-const (
-	AllowDotFiles  = false
-	IgnoreDotFiles = true
-)
+type Option func(l *Loader)
+
+func AllowDotFiles(loader *Loader)  { loader.ignoreDotfiles = false }
```
Comment (junr03) [inline loader/loader.go]: nit: do you want to use `l` instead of loader?

File: loader/loader.go [hunk 2]
```diff
@@ -157,12 +157,12 @@ func getFileSystemOp(ev fsnotify.Event) FileSystemOp {
...
+func IgnoreDotFiles(loader *Loader) { loader.ignoreDotfiles = true }
 
-func New(runtimePath string, runtimeSubdirectory string, scope stats.Scope, refresher Refresher, ignoreDotfiles bool) IFace {
+func New(runtimePath string, runtimeSubdirectory string, scope stats.Scope, refresher Refresher, opts ...Option) IFace {
```
Comment (junr03) [inline loader/loader.go]: Do you mind updating the README with the new paradigm?

**Comment by BenPope** [comment]
LGTM

**Comment by BenPope** [comment]
Thanks @twoismhttps://github.com/lyft/ratelimit/pull/52

**Comment by BenPope** [comment]
Thanks @twoism 

@junr03 https://github.com/lyft/ratelimit/pull/52

**Pull request merged**